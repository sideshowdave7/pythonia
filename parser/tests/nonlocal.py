def foo():
  x = 13
  print(x)
  
  def bar():
    nonlocal x
    print(x)
    x = 42
    print(x)
  
  bar()
  print(x)

foo()
