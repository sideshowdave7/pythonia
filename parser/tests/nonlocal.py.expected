(program
 (def
  (foo)
  (suite
   (= (x) 13)
   (expr (indexed print (called x)))
   (def
    (bar)
    (suite
     (nonlocal x)
     (expr (indexed print (called x)))
     (= (x) 42)
     (expr (indexed print (called x)))))
   (expr (indexed bar (called)))
   (expr (indexed print (called x)))))
 (expr (indexed foo (called))))

